File: vscode_extensions/shader-compiler/src/lsp_types_ext.ts
import { ProtocolRequestType } from "vscode-languageserver";

export interface ShaderPreviewParams {
    shaderFilePath: string;
}

export interface ShaderPreviewResult {
    code: string | null;
}

export const ShaderPreviewRequest =
    new ProtocolRequestType<ShaderPreviewParams, ShaderPreviewResult, never, void, void>("shader_compiler/ShaderPreview");

File: rs_shader_compiler_lsp/src/lsp_types_ext.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Default, Eq, PartialEq, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ShaderPreviewParams {
    pub shader_file_path: String,
}

#[derive(Debug, Default, Eq, PartialEq, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ShaderPreviewResult {
    pub code: Option<String>,
}

pub mod request {
    use super::{ShaderPreviewParams, ShaderPreviewResult};
    use lsp_types::request::Request;

    #[derive(Debug)]
    pub enum ShaderPreview {}

    impl Request for ShaderPreview {
        type Params = ShaderPreviewParams;
        type Result = ShaderPreviewResult;
        const METHOD: &'static str = "shader_compiler/ShaderPreview";
    }
}
